src/storage/mvcc/reader.rs

```diff
@@ -401,9 +406,153 @@ impl MvccReader {
+                if ts < commit_ts || &key != user_key {
+                    assert!(&key <= user_key);
+                    return self.get_value(user_key, lastest_version.0, lastest_version.1);
+                }
+                self.statistics.write.processed += 1;
+                Write::parse(self.write_cursor.as_mut().unwrap().value())?
+            };
+
+            match write.write_type {
+                WriteType::Put => {
+                    if write.short_value.is_some() {
+                        if self.key_only {
+                            lastest_version = (None, Some(vec![]));
+                        } else {
+                            lastest_version = (None, write.short_value.take());
+                        }
+                    } else {
+                        lastest_version = (Some(write.start_ts), None);
+                    }
+                }
+                WriteType::Delete => lastest_version = (None, None),
+                WriteType::Lock | WriteType::Rollback => {}
+            }
+            self.write_cursor
+                .as_mut()
+                .unwrap()
+                .prev(&mut self.statistics.write);
+        }
+
+        // After several prev, we still not get the latest version for the specified ts,
+        // use seek to locate the latest version.
+        let key = user_key.append_ts(ts);
+        assert!(self.write_cursor
+            .as_mut()
+            .unwrap()
+            .internal_seek(&key, &mut self.statistics.write)?);
+        loop {
+            let mut write = {
+                // If we reach the last handled key, it means we have check all versions
+                // for this user key.
+                if self.write_cursor.as_mut().unwrap().key()
+                    >= last_handled_key.as_ref().unwrap().as_slice()
+                {
+                    return self.get_value(user_key, lastest_version.0, lastest_version.1);
+                }
+
+                let w_cur = self.write_cursor.as_mut().unwrap();
+                let w_key = Key::from_encoded(w_cur.key().to_vec());
+                let commit_ts = w_key.decode_ts()?;
+                assert!(commit_ts <= ts);
+                let key = w_key.truncate_ts()?;
+                assert_eq!(&key, user_key);
+                self.statistics.write.processed += 1;
+                Write::parse(w_cur.value())?
+            };
+
+            match write.write_type {
+                WriteType::Put => {
+                    if write.short_value.is_some() {
+                        if self.key_only {
+                            return Ok(Some(vec![]));
+                        } else {
+                            return Ok(write.short_value.take());
+                        }
+                    } else {
+                        self.statistics.data.processed += 1;
+                        return Ok(Some(self.load_data(user_key, write.start_ts)?));
```

Comment:
when _short_value_ is _None_ and _key_only_ is _true_, we should not _load_data_.